### PR TITLE
Document privileged tokens

### PR DESCRIPTION
**Pull request by mziolekda**

### CODE REVIEW

File: docs/3.1/docs/app-dev/authorization.rst [hunk 1]
```diff
@@ -243,40 +243,17 @@ When authenticating as a user from a non-default identity provider configuration
 contain the ``iss`` field whose value matches the identity provider id.
 In case of the default identity provider configuration, the ``iss`` field can be empty or omitted from the access tokens.
 
+Privileged Tokens
+=================
 
-Custom Daml Claims Access Tokens
```
Comment (mziolekda) [inline docs/3.1/docs/app-dev/authorization.rst:247]: Custom tokens are removed in the 3.x release line

File: docs/3.1/docs/app-dev/authorization.rst [hunk 2]
```diff
@@ -243,40 +243,17 @@ When authenticating as a user from a non-default identity provider configuration
...
-- ``applicationId`` requires requests with this token to use that application id or not set an application id at all, which should be used to distinguish requests from different applications
-- ``exp`` is the standard JWT expiration date (in seconds since EPOCH)
-- ``actAs``, ``readAs`` and (participant) ``admin`` encode the rights granted by this access token
-
-The ``public`` right is implicitly granted to any request bearing a non-expired JWT issued by a trusted issuer with matching ``ledgerId``, ``participantId`` and ``applicationId`` values.
+A Privileged Token allows the caller to gain either ``wildcard-`` or ``participant-admin-`` level access. A ``wildcard-``level access combines
+the capabilities of the ``participant-admin`` with the right to ``act-as`` and ``read-as`` any party. Privileged tokens look like and are
+parsed as any other token. They can follow both the audience-based and the scope-based format. Unlike for the User Tokens, the entry specified
+in the ``sub`` field need not represent a user existing in the participant's database. In fact this field is not validated nor checked at all.
```
Comment (carrielaben-da) [inline docs/3.1/docs/app-dev/authorization.rst:252]:
```suggestion
parsed as any other token. They can follow both the audience-based and the scope-based format. Unlike with user tokens, the entry specified
in the ``sub`` field is not validated or checked and need not represent a user existing in the participant's database. 
```

File: docs/3.1/docs/app-dev/authorization.rst [hunk 3]
```diff
@@ -243,40 +243,17 @@ When authenticating as a user from a non-default identity provider configuration
...
 
-.. note:: All Daml ledgers also support a deprecated legacy format of custom Daml claims
-   access tokens whose format is equal to the above except that the custom claims
-   are present at the same level as ``exp`` in the token above,
-   instead of being nested below ``"https://daml.com/ledger-api"``.
+The use of Privileged Tokens is disabled by default and has to be explicitly enabled in the Canton configuration. Unlike normal
+User Tokens, the Privileged Tokens can be configured also for the Admin APIs exposed by the participant, the sequencer and
+the mediator nodes.
```
Comment (carrielaben-da) [inline docs/3.1/docs/app-dev/authorization.rst:256]:
```suggestion
Use of privileged tokens is inactive by default and must be explicitly enabled in the Canton configuration. Unlike normal
user tokens, privileged tokens can also be configured for the Admin APIs exposed by the participant, the sequencer, and
the mediator nodes.
```

File: docs/3.1/docs/app-dev/authorization.rst [hunk 4]
```diff
@@ -243,40 +243,17 @@ When authenticating as a user from a non-default identity provider configuration
...
-
-where all of the fields are optional, and if present,
-
-- ``ledgerId`` and ``participantId`` restrict the validity of the token to the given ledger or participant node
-- ``applicationId`` requires requests with this token to use that application id or not set an application id at all, which should be used to distinguish requests from different applications
-- ``exp`` is the standard JWT expiration date (in seconds since EPOCH)
-- ``actAs``, ``readAs`` and (participant) ``admin`` encode the rights granted by this access token
-
-The ``public`` right is implicitly granted to any request bearing a non-expired JWT issued by a trusted issuer with matching ``ledgerId``, ``participantId`` and ``applicationId`` values.
+A Privileged Token allows the caller to gain either ``wildcard-`` or ``participant-admin-`` level access. A ``wildcard-``level access combines
```
Comment (carrielaben-da) [inline docs/3.1/docs/app-dev/authorization.rst:249]:
```suggestion
A privileged token allows the caller to gain either ``wildcard-`` or ``participant-admin-`` level access. A ``wildcard-``level access combines
```

File: docs/3.1/docs/canton/usermanual/apis.rst [hunk 5]
```diff
@@ -243,6 +243,17 @@ Other scopes can be configured explicitly using the custom target scope configur
 Target scope can be any case-sensitive string containing alphanumeric characters, hyphens, slashes, colons and underscores.
 Either the ``target-scope`` or the ``target-audience`` parameter can be configured individually, but not both at the same time.
 
+Configuring Privileged Tokens
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Support for privileged tokens can be configured for the Ledger API exposed by the participant node as well as the Admin API exposed
+by the participant, the sequencer and the mediator nodes. The configuration follows the same pattern as for the configuration of
```
Comment (carrielaben-da) [inline docs/3.1/docs/canton/usermanual/apis.rst:250]:
```suggestion
by the participant, the sequencer, and the mediator nodes. The configuration follows the same pattern as for the configuration of
```

File: docs/3.1/docs/canton/usermanual/apis.rst [hunk 6]
```diff
@@ -243,6 +243,17 @@ Other scopes can be configured explicitly using the custom target scope configur
...
+normal user tokens but with an addition of a key called ``privileged`` set to true.
+
+Additionally, it is possible to determine if the use of privileged tokens should result in granting of the ``participant_admin``
+or the ``wildcard`` access levels. This can be achieved by adding a definition of the ``access-level`` key and setting it to
+``Admin`` or ``Wildcard`` respectively. The ``Admin`` is the default.
```
Comment (carrielaben-da) [inline docs/3.1/docs/canton/usermanual/apis.rst:255]:
```suggestion
Additionally, you can determine if the use of privileged tokens should result in granting of the ``participant_admin``
or the ``wildcard`` access levels by adding a definition of the ``access-level`` key and setting it to
``Admin`` or ``Wildcard`` respectively. The ``Admin`` is the default.
```

**Review by carrielaben-da — APPROVED** [review]

**Review by mziolekda — COMMENTED** [review]
Thank you @yves-da

**Pull request merged**